addons/payment_paypal/models/payment_transaction.py

```diff
@@ -0,0 +1,100 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import logging
+
+from werkzeug import urls
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+from odoo.addons.payment import utils as payment_utils
+from odoo.addons.payment_paypal.controllers.main import PaypalController
+
+_logger = logging.getLogger(__name__)
+
+
+class PaymentTransaction(models.Model):
+    _inherit = 'payment.transaction'
+
+    # See https://developer.paypal.com/docs/api-basics/notifications/ipn/IPNandPDTVariables/
+    paypal_type = fields.Char(string="PayPal Transaction Type")
+
+    def _get_specific_rendering_values(self, processing_values):
+        if self.acquirer_id.provider != 'paypal':
+            return super()._get_specific_rendering_values(processing_values)
+
+        base_url = self.acquirer_id._get_base_url()
+        partner_first_name, partner_last_name = payment_utils.split_partner_name(self.partner_name)
+        notify_url = self.acquirer_id.paypal_use_ipn \
+                     and urls.url_join(base_url, PaypalController._notify_url)
+        return {
+            **processing_values,
+            'address1': self.partner_address,
+            'business': self.acquirer_id.paypal_email_account,
+            'city': self.partner_city,
+            'cmd': '_xclick',
+            'country': self.partner_country_id.code,
+            'currency_code': self.currency_id.name,
+            'email': self.partner_email,
+            'first_name': partner_first_name,
+            'handling': self.acquirer_id.fees_active and self.fees,
```

Comment:
I'm not sure we should check at this step if the acquirer has fees enabled.  If the transaction has fees, it means that the acquirer had fees enabled when the transaction was created, and we have to show it, even if the configuration of the acquirer changed no?

Reply:
Indeed.